packages/flutter_tools/lib/src/platform_plugins.dart

```diff
@@ -84,33 +84,45 @@ class AndroidPlugin extends PluginPlatform {
       'src',
       'main',
     );
-    File mainPluginClass = fs.file(
+
+    final List<String> mainClassCandidates = <String>[
       fs.path.join(
         baseMainPath,
         'java',
         package.replaceAll('.', fs.path.separator),
         '$pluginClass.java',
+      ),
+      fs.path.join(
+        baseMainPath,
+        'kotlin',
+        package.replaceAll('.', fs.path.separator),
+        '$pluginClass.kt',
       )
-    );
-    // Check if the plugin is implemented in Kotlin since the plugin's pubspec.yaml
-    // doesn't include this information.
-    if (!mainPluginClass.existsSync()) {
-      mainPluginClass = fs.file(
-        fs.path.join(
-          baseMainPath,
-          'kotlin',
-          package.replaceAll('.', fs.path.separator),
-          '$pluginClass.kt',
-        )
+    ];
+
+    File mainPluginClass;
+    bool mainClassFound = false;
+    for (String mainClassCandidate in mainClassCandidates) {
+      mainPluginClass = fs.file(mainClassCandidate);
+      if (mainPluginClass.existsSync()) {
+        mainClassFound = true;
+        break;
+      }
+    }
+    if (!mainClassFound) {
+      throwToolExit(
+        'The plugin $name doesn\'t have a main class defined in ${mainClassCandidates.join(' or ')}. '
+        'This is likely to due to an incorrect `androidPackage: $package` entry in the plugin\'s pubspec.yaml. '
+        'Please contact the author of this plugin and consider using a different plugin in the meanwhile. '
```

Comment:
The person getting the error is most likely to be the plugin author themselves. We should make sure to give useful actionable information to the plugin author as well.

Reply:
sounds good. I added this to the message: "If you are the author of this plugin, fix the `androidPackage` entry or move the main class to any of locations used above."